aten/src/ATen/native/Pow.cpp

```diff
@@ -27,11 +27,8 @@ Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
            "result type ", common_dtype, "can't be cast to the desired output type ",
            result.scalar_type());
 
-  if (exp.isComplex() && (exp.toComplexDouble() == 0.0) ) {
-    result.resize_as_(base).fill_(1);
-  } else if (exp.isComplex() && (exp.toComplexDouble() == 1.0) ) {
-    result.resize_as_(base).fill_(base);
-  } else if (!exp.isComplex() && (exp.toDouble() == 0.0)) {
+  auto exponent = (exp.isComplex()) ? exp.toComplexDouble() : exp.toDouble();
+  if (exponent == 0.0) {
     result.resize_as_(base).fill_(1);
   } else if (!exp.isComplex() && (exp.toDouble() == 1.0)) {
```

Comment:
This should be updated as well no? otherwise the case for complex == 1 is not handles anymore.